psd-web/app/controllers/collaborators_controller.rb

```diff
@@ -35,11 +34,36 @@ def create
     end
   end
 
+  def edit
+    authorize @investigation, :add_collaborators?
+
+    @team = Team.find(params[:id])
+    @collaborator = @investigation.collaborators.find_by! team_id: @team.id
```

Comment:
suggestion: use the collaborator id in the path, and then look it up using:

```suggestion
    @collaborator = Collaborator.find(params[:id])
    @team = @collaborator.team
```

Reply:
Is there any value over current approach?

Reply:
No. And actually your way checks that the two id params match, so is probably better.